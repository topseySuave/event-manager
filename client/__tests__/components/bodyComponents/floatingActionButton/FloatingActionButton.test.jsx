import React from 'react';
import { shallow } from 'enzyme';
import { FloatingActionButton, mapStateToProps } from
  '../../../../src/components/bodyComponents/floatingActionButton/FloatingActionButton'; // eslint-disable-line

let wrapper, instance, mounted;
let store, componentProps;

const getComponent = (props) => {
  mounted = shallow(<FloatingActionButton {...props} />);
  return mounted;
};

const componentWillMountSpy = jest
  .spyOn(FloatingActionButton.prototype, 'componentWillMount');

describe('FloatingActionButton component', () => {
  test('should mount', () => {
    componentProps = {
      activeState: {
        isAuthenticated: true,
        user: { id: 1, role: true }
      }
    };
    wrapper = getComponent(componentProps);
    expect(wrapper).toMatchSnapshot();
    expect(componentWillMountSpy).toBeCalled();
  });

  test('should render nothing when user not authenticated', () => {
    componentProps = {
      activeState: {
        isAuthenticated: false
      }
    };
    wrapper = getComponent(componentProps);
    mapStateToProps({});
    expect(wrapper.html()).toBeFalsy();
  });
});
