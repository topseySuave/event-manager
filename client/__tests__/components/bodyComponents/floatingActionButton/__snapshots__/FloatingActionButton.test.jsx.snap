// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingActionButton component FloatingActionButton component should mount 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FloatingActionButton
    activeState={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "id": 1,
          "role": true,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="tooltipped modal-trigger btn-largebtn-floating pulse green lighten-1"
        data-position="top"
        data-tooltip="Add Center"
        href="#edit_center_modal"
      >
        <i
          className="material-icons"
        >
          add
        </i>
      </a>,
      "className": "fixed-action-btn wow zoomIn",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <i
          className="material-icons"
        >
          add
        </i>,
        "className": "tooltipped modal-trigger btn-largebtn-floating pulse green lighten-1",
        "data-position": "top",
        "data-tooltip": "Add Center",
        "href": "#edit_center_modal",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "add",
          "className": "material-icons",
        },
        "ref": null,
        "rendered": "add",
        "type": "i",
      },
      "type": "a",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="tooltipped modal-trigger btn-largebtn-floating pulse green lighten-1"
          data-position="top"
          data-tooltip="Add Center"
          href="#edit_center_modal"
        >
          <i
            className="material-icons"
          >
            add
          </i>
        </a>,
        "className": "fixed-action-btn wow zoomIn",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="material-icons"
          >
            add
          </i>,
          "className": "tooltipped modal-trigger btn-largebtn-floating pulse green lighten-1",
          "data-position": "top",
          "data-tooltip": "Add Center",
          "href": "#edit_center_modal",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "add",
            "className": "material-icons",
          },
          "ref": null,
          "rendered": "add",
          "type": "i",
        },
        "type": "a",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
